internal/telemetry/event.go

```diff
@@ -259,6 +259,16 @@ func withProjectID(cmd CmdFlags, c ProjectIDGetter) EventOpt {
 	}
 }
 
+type SourceGetter interface {
+	Source() string
+}
+
+func withUserType() EventOpt {
+	return func(event Event) {
+		event.Properties["user_type"] = config.UserType
```

Comment:
the pattern we have for env vars in the same as `withCI` where we check if the env var is present and set the property to true, I don't think we need the concept of `user_type`

Reply:
Are you suggesting setting the CI property of telemetry to `university`?

Reply:
I think that regardless of the env var we ask mdb university to use, the telemetry property should be an enum, so we can add more values in the future. So either `cli_source` or `user_type`.
For the university we have 2 options:
1. MONGODB_UNIVERSITY_USER=true env var (and translate to cli_source or user_type = university)
2. CLI_SOURCE or USER_TYPE=university and just pass to the telemetry as a property with the same name
@fmenezes was your suggestion more in line with second option based on current telemetry implementation?

Reply:
I'm suggesting to use the env var key as the property

For instance mongodb-university-user=true/false

Reply:
mh.. This solution will force us to add a new telemetry property instead of updating just the enum if we find another user type in the future. @tibulca WDYT?

Reply:
@andreaangiolillo are you expecting any other user types?

Reply:
I _personally_ think user_type is quite confusing especially because it does not make me remember the original env var.

I feel telemetry should be as close as possible to the running environment.

If our env var is called `MONGODB_UNIVERSITY_USER` _I think_ we should forward _exactly that_ to telemetry.

Reply:
understood. I am wondering if renaming `MONGODB_UNIVERSITY_USER_TYPE` remove the confusion 🤔 I don't have a strong opinion here. I am happy to add `MONGODB_UNIVERSITY_USER` as a telemetry prop. I will wait for Ciprian to confirm and then I will address the comment. Thanks for the feedback! 🙇

Reply:
If it's keeping the same name for the env var and then option 2 works, right?
```
CLI_SOURCE or USER_TYPE=university and just pass to the telemetry as a property with the same name
@fmenezes was your suggestion more in line with second option based on current telemetry implementation?
```
There's no other source currently (as this is the first use case), but I don't think we should exclude that for the future and end up with another MONGODB_SOME_OTHER_TYPE and so on

Reply:
@tibulca it works for me

Reply:
in telemetry there is a source property currently at https://github.com/mongodb/mongodb-atlas-cli/blob/594c4fe2ddfc1f0586a5d8d007076c76efe053fd/internal/telemetry/event.go#L332

Reply:
I wouldn't touch it as it would require mms code changes

Reply:
Decision:
Use `CLI_USER_TYPE =university` and add a new telemetry prop with the same name as the env variable

Reply:
I updated the PR with the new env variable and telemetry property.

> the telemetry property should be an enum

Regarding this comment, golang doesn't support enums. I don't think that simulating an enum with iota here would improve the usability of the code. WDYT?